src/libcore/option.rs

```diff
@@ -183,6 +183,20 @@ impl<T> Option<T> {
     /// ```
     ///
     /// [`Some`]: #variant.Some
+    #[rustc_const_unstable(feature = "const_option_match")]
+    #[cfg(not(bootstrap))]
+    #[must_use = "if you intended to assert that this has a value, consider `.unwrap()` instead"]
+    #[inline]
+    #[stable(feature = "rust1", since = "1.0.0")]
+    pub const fn is_some(&self) -> bool {
+        match *self {
+            Some(_) => true,
+            None => false,
+        }
+    }
```

Comment:
Why do you duplicate all these function bodies?

Reply:
Because you can't conditional compilation add the const modifier to the function. So you need one function for bootstrap (without the const control flow feature and thus without the function being const) and one for not bootstrap

Reply:
Oh I see.

Alternatively we could wait a week until bootstrap gets bumped, that would make everything much easier to review too and avoid all the churn.